packages/editor/src/client/blocks/richTextBlock/toolbar/linkMenu.tsx

```diff
@@ -0,0 +1,168 @@
+import React, {useState, useEffect, useContext} from 'react'
+import {useTranslation} from 'react-i18next'
+import {Button, Form, FormGroup, ControlLabel, FormControl, ButtonToolbar} from 'rsuite'
+import {Transforms, Range, Editor} from 'slate'
+import {useSlate} from 'slate-react'
+import {WepublishEditor} from '../editor/wepublishEditor'
+import {InlineFormat} from '../editor/formats'
+import {SubMenuContext} from '../../../atoms/toolbar'
+import axios from 'axios'
+
+export function LinkMenu() {
+  const editor = useSlate()
+
+  const {closeMenu} = useContext(SubMenuContext)
+
+  const [title, setTitle] = useState('')
+  const [url, setURL] = useState('')
+
+  const [selection, setSelection] = useState<Range | null>(null)
+
+  const [isValidURL, setIsValidURL] = useState(false)
+  const isDisabled = !isValidURL
+
+  const {t} = useTranslation()
+
+  useEffect(() => {
+    validateURL(url).then(value => setIsValidURL(value))
+  }, [url])
+
+  useEffect(() => {
+    setSelection(editor.selection)
+
+    const nodes = Array.from(
+      WepublishEditor.nodes(editor, {
+        at: editor.selection ?? undefined,
+        match: node => node.type === InlineFormat.Link
+      })
+    )
+    const tuple = nodes[0]
+    if (tuple) {
+      const [node] = tuple
+      setTitle((node.title as string) ?? '')
+      setURL((node.url as string) ?? '')
+    } else if (editor.selection) {
+      const text = Editor.string(editor, editor.selection)
+      setTitle(text ?? '')
+    }
+  }, [])
+
+  return (
+    <>
+      <Form fluid>
+        <FormGroup>
+          <ControlLabel>{t('blocks.richText.link')}</ControlLabel>
+          <FormControl
+            errorMessage={url && !isValidURL ? 'Invalid Link' : undefined}
+            value={url}
+            onChange={url => setURL(url)}
+          />
+        </FormGroup>
+        <FormGroup>
+          <ControlLabel>{t('blocks.richText.text')}</ControlLabel>
+          <FormControl
+            value={title}
+            onChange={title => {
+              setTitle(title)
+            }}
+          />
+        </FormGroup>
+        <ButtonToolbar>
+          <Button
+            disabled={isDisabled}
+            onClick={e => {
+              e.preventDefault()
+              insertLink(editor, selection, url, title || undefined)
+              closeMenu()
+            }}>
+            {t('blocks.richText.insert')}
+          </Button>
+          <RemoveLinkFormatButton />
+        </ButtonToolbar>
+      </Form>
+    </>
+  )
+}
+
+async function validateURL(url: string) {
+  if (url) {
+    const pattern = new RegExp(
+      '^(https?:\\/\\/)?' + // protocol
+      '((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|' + // domain name
+      '((\\d{1,3}\\.){3}\\d{1,3}))' + // OR ip (v4) address
+      '(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*' + // port and path
+      '(\\?[;&a-z\\d%_.~+=-]*)?' + // query string
+        '(\\#[-a-z\\d_]*)?$',
+      'i'
+    )
+    if (!pattern.test(url)) return false
```

Comment:
![deepin-screen-recorder_Select area_20210217103534](https://user-images.githubusercontent.com/2340416/108184863-f988c080-710b-11eb-93e2-ebaf4651ef39.gif)

My URL with `https` in the begging was invalid.